sdk/nodejs/cmd/dynamic-provider/index.ts

```diff
@@ -56,18 +58,36 @@ process.on("exit", (code: number) => {
 
 const providerCache: { [key: string]: dynamic.ResourceProvider } = {};
 
-function getProvider(props: any): dynamic.ResourceProvider {
+// getProvider deserializes the provider from the string found in
+// `props[providerKey]` and calls `provider.configure` with the config. The
+// deserialized and configured provider is stored in `providerCache`. This
+// guarantees that the provider is only deserialized and configured once per
+// process.
```

Comment:
`/**` / `*/` for TypeDoc. I don't think we'll need `@internal` since we're not exporting, but that might apply elsewhere.